tests/system/common.py

```diff
@@ -790,3 +790,26 @@ def task_by_name(tasks, name):
 
 async def assert_event(event_type, event_stream, within=10):
     await asyncio.wait_for(find_event(event_type, event_stream), within)
+
+
+def kill_process_on_host(hostname, pattern):
+    """ Kill the process matching pattern at ip
+
+        :param hostname: the hostname or ip address of the host on which the process will be killed
+        :param pattern: a regular expression matching the name of the process to kill
+        :return: True if at least one process gets killed, or False otherwise
+    """
+
+    status, stdout = shakedown.run_command_on_agent(hostname, "ps aux | grep -v grep | grep '{}'".format(pattern))
+    pids = [p.strip().split()[1] for p in stdout.splitlines()]
+    killed_some = False
+
+    for pid in pids:
+        status, stdout = shakedown.run_command_on_agent(hostname, "sudo kill -9 {}".format(pid))
```

Comment:
Ideally the `ps` and the kills should run within one command to reduce the number of SSH requests.

Reply:
Yep, the should totally just be a one-liner. It seems that Python doesn't need to be involved.

Reply:
It is an amended version of `shakedown.kill_process_on_host`, which returns `None`, so there is no way to tell if anything has been killed at all. I agree that it can be improved.